front/dashboard/src/app/components/filtro/filtro.component.ts

```diff
@@ -12,8 +12,29 @@ export class FiltroComponent {
   tipos_documento: any
   buttonsDoc: any
   buttonsEmp: any
+  segmentos: any
 
+  async Filterseg(){
+    let res = await getUsers();
+    this.empresas = res.data.user;
+  
+    if (this.empresas && Array.isArray(this.empresas) && this.empresas.length > 0) {
+      this.segmentos = [];
+  
+      for (let i = 0; i < this.empresas.length; i++) {
+        if (this.empresas[i].segment) {
+          this.segmentos = this.segmentos.concat(this.empresas[i].segment);
+        }
+      }
+
+      this.segmentos = Array.from(new Set(this.segmentos));
+  
+      console.log(this.segmentos);
+    }
```

Comment:
Acho que se usar a outra sintaxe do for e usar Array.push ao invés de Array.concat esse pedaço de código pode ficar bem menor.

```suggestion
    this.segmentos = []
    
    for (let empresa of this.empresas) {
        this.segmentos.push(empresa.segment)
    }
```